File: core/src/net/bossdragon/util/ai/LocationableEntity.java
package net.bossdragon.util.ai;

import com.badlogic.gdx.ai.utils.Location;
import com.badlogic.gdx.math.Vector2;
import net.bossdragon.component.base.Position;
import net.bossdragon.component.base.Velocity;

/**
 *
 */
public class LocationableEntity implements Location<Vector2> {
    public final Position pos;
    public final Velocity vel;

    public LocationableEntity(final Position pos, final Velocity vel) {
        this.pos = pos;
        this.vel = vel;
    }

    @Override
    public Vector2 getPosition() {
        return pos.currentPos;
    }

    @Override
    public float getOrientation() {
        return 0;
    }

    @Override
    public void setOrientation(float orientation) {

    }

    @Override
    public float vectorToAngle(Vector2 vector) {
        return 0;
    }

    @Override
    public Vector2 angleToVector(Vector2 outVector, float angle) {
        return null;
    }

    @Override
    public Location<Vector2> newLocation() {
        return null;
    }
}

File: core/src/net/bossdragon/util/ai/LocationableEntity.kt
package net.bossdragon.util.ai

import com.badlogic.gdx.ai.utils.Location
import com.badlogic.gdx.math.Vector2
import net.bossdragon.component.base.Position
import net.bossdragon.component.base.Velocity

/**
 *
 */
class LocationableEntity(val pos: Position, val vel: Velocity) : Location<Vector2> {


    override fun setOrientation(orientation: Float) {

    }

    override fun vectorToAngle(vector: Vector2): Float {
        TODO("not implemented")
    }

    override fun getPosition(): Vector2 {
        return pos.currentPos
    }

    override fun angleToVector(outVector: Vector2, angle: Float): Vector2 {
        TODO("not implemented")
    }

    override fun newLocation(): Location<Vector2> {
        TODO("not implemented")
    }

    override fun getOrientation(): Float {
        // TODO
        return 0f
    }

}
